app/src/main/java/com/github/mik629/android/fundamentals/ui/moviedetails/FragmentMovieDetails.kt

```diff
@@ -1,90 +1,104 @@
 import com.github.mik629.android.fundamentals.BuildConfig
-import com.github.mik629.android.fundamentals.GlideApp
 import com.github.mik629.android.fundamentals.R
 import com.github.mik629.android.fundamentals.databinding.FragmentMovieDetailsBinding
-import com.github.mik629.android.fundamentals.domain.model.MovieItem
+import com.github.mik629.android.fundamentals.di.AppModule
+import com.github.mik629.android.fundamentals.di.buildGlideRequest
 import com.github.mik629.android.fundamentals.ui.global.ActorItemAdapter
+import com.github.mik629.android.fundamentals.ui.utils.setRating
+import com.google.android.material.snackbar.BaseTransientBottomBar
+import com.google.android.material.snackbar.Snackbar
 
-class FragmentMovieDetails : Fragment() {
-    private lateinit var binding: FragmentMovieDetailsBinding
+class FragmentMovieDetails : Fragment(R.layout.fragment_movie_details) {
+    private val binding by viewBinding(FragmentMovieDetailsBinding::bind)
 
     private val glideRequest by lazy {
-        GlideApp.with(this)
-            .asDrawable()
-            .thumbnail(0.1f)
-            .placeholder(R.drawable.ic_image_loading)
-            .fallback(R.drawable.ic_broken_image)
-            .transition(DrawableTransitionOptions.withCrossFade())
-            .fitCenter()
+        buildGlideRequest(this)
     }
+    private val actorItemAdapter by lazy {
+        ActorItemAdapter(glideRequest)
+    }
+
+    private var movieId: Long = 0
 
-    private var movieItem: MovieItem? = null
+    private val viewModel: MovieDetailsViewModel by viewModels(
+        factoryProducer = {
+            MovieDetailsViewModelFactory(requireContext(), movieId)
+        }
+    )
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        movieItem = (savedInstanceState ?: arguments)?.getParcelable(ARG_MOVIE_ITEM)
         super.onCreate(savedInstanceState)
+        movieId = arguments?.getLong(ARG_MOVIE_ID) ?: 0
     }
 
-    override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View {
-        binding = FragmentMovieDetailsBinding.inflate(layoutInflater)
-
-        with(binding) {
-            back.setOnClickListener {
-                requireFragmentManager().popBackStack()
-            }
-
-            movieItem?.let {
-                age.text = getString(R.string.movie_min_age, it.minAge)
-                movieTitle.text = it.title
-                genre.text = it.genres.joinToString { genre -> genre.name }
-                ratingLayout.ratingBar.rating = it.rating
-                storyline.text = it.overview
-                reviews.text = getString(R.string.movie_reviews, it.reviews)
-                val actorItemAdapter = ActorItemAdapter(glideRequest)
-                actors.adapter = actorItemAdapter
+    override fun onStart() {
+        super.onStart()
+        viewModel.movieDetails.observe(viewLifecycleOwner) {
```

Comment:
observe in onStart ?
try to hide app(to onStop be called)few times and calculate how many times movieItemAdapter.submitList(it) was called in onStart